app/MMB/src/scenes/tickets/boardingPasses/appleWallet/AppleWalletPassenger.js

```diff
@@ -9,14 +9,18 @@ import { withNavigation } from 'react-navigation';
 import type { NavigationType } from '@kiwicom/mobile-navigation';
 
 import type { AppleWalletPassenger as AppleWalletType } from './__generated__/AppleWalletPassenger.graphql';
+import WalletContext from './../../../../context/WalletContext';
 
-type Props = {|
-  +data: AppleWalletType,
-  +navigation: NavigationType,
+type PropsWithContext = {|
+  ...Props,
+  addPkpassData: (passengerName: string, pkpassUrl: string, id: string) => void,
 |};
 
-class AppleWalletPassenger extends React.Component<Props> {
+class AppleWalletPassenger extends React.Component<PropsWithContext> {
   onPress = () => {
+    const name = idx(this.props.data, _ => _.passenger.fullName) || '';
+    const url = idx(this.props.data, _ => _.url) || '';
+    this.props.addPkpassData(name, url, this.props.legId || '');
```

Comment:
This won't do anything if `this.props.legId` is `null` or `undefined`, right? So maybe I would add a conditional instead of some default empty string value because there is no reason to call that function in that case. Similarly, you could add the variable creations after checking for legId.

Reply:
yes that makes sense